### PR TITLE
CompanyService_GetCompanyById

### PR DESCRIPTION
**Pull request by KostinP**
It was necessary to add the ability to get company by id. For this, I added:
- Repository
- Request model
- Mapper
- Command
- Controller
- Tests

### CODE REVIEW

File: DigitalOffice/Services/CompanyService/CompanyService.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     </PropertyGroup>
 
     <ItemGroup>
+      <PackageReference Include="FluentValidation" Version="9.0.1" />
```
Comment (lalagvana) [inline DigitalOffice/Services/CompanyService/CompanyService.csproj]: No need

File: DigitalOffice/Services/CompanyService/Controllers/CompanyController.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System;
+
+namespace LT.DigitalOffice.CompanyService.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CompanyController : ControllerBase
+    {
+        [HttpPost("getCompanyById")]
+        public Company GetCompanyById([FromServices] IGetCompanyByIdCommand command, [FromRoute] Guid companyId)
```
Comment (lalagvana) [inline DigitalOffice/Services/CompanyService/Controllers/CompanyController.cs]: Not FromQuery?

File: DigitalOffice/Services/CompanyService/Commands/Interfaces/IGetCompanyByIdCommand.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using LT.DigitalOffice.CompanyService.Models;
+using System;
+
+namespace LT.DigitalOffice.CompanyService.Commands.Interfaces
+{
+    public interface IGetCompanyByIdCommand
```
Comment (MihailKulikov) [inline DigitalOffice/Services/CompanyService/Commands/Interfaces/IGetCompanyByIdCommand.cs:10]: XML documentation

Comment (MihailKulikov) [inline DigitalOffice/Services/CompanyService/Commands/Interfaces/IGetCompanyByIdCommand.cs:10]: XML documentation

File: DigitalOffice/Services/CompanyService/Mappers/Interfaces/IMapper.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace LT.DigitalOffice.CompanyService.Mappers.Interfaces
+{
+    public interface IMapper<in TIn, out TOut>
```
Comment (MihailKulikov) [inline DigitalOffice/Services/CompanyService/Mappers/Interfaces/IMapper.cs]: XML documentation

File: DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        }
+
+        [Test]
+        public void SuccessfulGetCompanyById()
+        {
+            var result = repository.GetCompanyById(dbCompanyInDB.Id);
+
+            var resultCompany = dbContext.Companies.FirstAsync().Result;
+            SerializerAssert.AreEqual(result, resultCompany);
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs]: Two results? Rename variables and swap arguments

File: DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+        [TearDown]
+        public void Clear()
+        {
+            if (dbContext.Database.IsInMemory())
+            {
+                dbContext.Database.EnsureDeleted();
+            }
+        }
+
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs:77]: Add test that checks throwing exception if there are no such company in db.

**Review by MihailKulikov — CHANGES_REQUESTED** [review]

File: DigitalOffice/Services/CompanyService/CompanyService.csproj (hunk 1 above)
Comment (KostinP) [reply DigitalOffice/Services/CompanyService/CompanyService.csproj]: Fixed.

File: DigitalOffice/Services/CompanyService/Controllers/CompanyController.cs (hunk 2 above)
Comment (KostinP) [reply DigitalOffice/Services/CompanyService/Controllers/CompanyController.cs]: Fixed.

File: DigitalOffice/Services/CompanyService/Commands/Interfaces/IGetCompanyByIdCommand.cs (hunk 3 above)
Comment (KostinP) [reply DigitalOffice/Services/CompanyService/Commands/Interfaces/IGetCompanyByIdCommand.cs:10]: Fixed.

Comment (KostinP) [reply DigitalOffice/Services/CompanyService/Commands/Interfaces/IGetCompanyByIdCommand.cs:10]: Fixed.

File: DigitalOffice/Services/CompanyService/Mappers/Interfaces/IMapper.cs (hunk 4 above)
Comment (KostinP) [reply DigitalOffice/Services/CompanyService/Mappers/Interfaces/IMapper.cs]: Fixed.

File: DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs (hunk 5 above)
Comment (KostinP) [reply DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs]: Fixed.

File: DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs (hunk 6 above)
Comment (KostinP) [reply DigitalOffice/Tests/CompanyServiceTests/Repositories/CompanyRepositoryTests.cs:77]: Fixed.

**Review by MihailKulikov — APPROVED** [review]

**Review by ifelseelif — APPROVED** [review]

**Review by mekk1t — APPROVED** [review]

**Review by lalagvana — APPROVED** [review]

File: DigitalOffice/Tests/CompanyServiceTests/Mappers/CompanyMapperTests.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        [Test]
+        public void ThrowExceptionIfArgumentIsNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => mapper.Map(null));
+        }
+
+        [Test]
+        public void ReturnRightModelSuccessfully()
```
Comment (lalagvana) [inline DigitalOffice/Tests/CompanyServiceTests/Mappers/CompanyMapperTests.cs]: RightModel?

File: DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+        [Test]
+        public void FailGetCompanyAccordingToRepository()
+        {
+            repositoryMock
+                .Setup(x => x.GetCompanyById(It.IsAny<Guid>()))
+                .Throws(new Exception());
+
+            mapperMock
+                .Setup(X => X.Map(It.IsAny<DbCompany>()))
```
Comment (lalagvana) [inline DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs]: Why big X?

File: DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+        [Test]
+        public void SuccessfulGetCompany()
+        {
+            repositoryMock
+                .Setup(x => x.GetCompanyById(It.IsAny<Guid>()))
+                .Returns(dbCompany);
+
+            mapperMock
+                .Setup(X => X.Map(It.IsAny<DbCompany>()))
```
Comment (lalagvana) [inline DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs]: Why big X?

**Review by MihailKulikov — CHANGES_REQUESTED** [review]
don't forget about controller tests :)
Resolve conflicts

File: DigitalOffice/Tests/CompanyServiceTests/Mappers/CompanyMapperTests.cs (hunk 7 above)
Comment (KostinP) [reply DigitalOffice/Tests/CompanyServiceTests/Mappers/CompanyMapperTests.cs]: Fixed names.

File: DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs (hunk 8 above)
Comment (KostinP) [reply DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs]: Fixed.

File: DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs (hunk 9 above)
Comment (KostinP) [reply DigitalOffice/Tests/CompanyServiceTests/Commands/GetCompanyByIdCommandTests.cs]: Fixed.

**Review by MihailKulikov — APPROVED** [review]
(not necessary) remove unused directives in test files

**Review by djino97 — APPROVED** [review]

**Pull request merged**